src/app/public/modules/datepicker/datepicker-input.directive.ts

```diff
@@ -333,21 +333,14 @@ export class SkyDatepickerInputDirective
 
     const value = event.target.value;
 
-    let errors: ValidationErrors;
-
-    if (this.isDateStringValid(value)) {
-      /* tslint:disable:no-null-keyword */
-      errors = null;
```

Comment:
If I type an invalid string (such as `'asdf 3'`) it invalidates the field, which is good, but if I keep focus in the input and retype a _valid_ string (such as `'3/3/2012'`) the field only becomes valid when I blur the input (with the tab key). I originally added `errors = null;` to make the field valid when the user corrected the string while typing. Maybe you could add some logic in here to only make the field valid if no other errors exist on the input?